src/MakePIAPortable/MakePIAPortable.csproj

```diff
@@ -26,35 +26,69 @@
+    <ILAsmFileLocation Condition="$([MSBuild]::IsOSPlatform('linux'))">$(Pkgruntime_linux-x64_Microsoft_NETCore_ILAsm)\runtimes\linux-x64\native\ilasm</ILAsmFileLocation>
+    <ILAsmFileLocation Condition="$([MSBuild]::IsOSPlatform('osx'))">$(Pkgruntime_osx-x64_Microsoft_NETCore_ILAsm)\runtimes\osx-x64\native\ilasm</ILAsmFileLocation>
+    
+    <MakePIAPortableExe>MakePIAPortable.exe</MakePIAPortableExe>
+    <!-- Some machines might not have current directory in the PATH, force linux machines to run from current directory. -->
+    <MakePIAPortableExe Condition="$([MSBuild]::IsOsPlatform('linux'))">./MakePIAPortable.exe</MakePIAPortableExe>
+    <ILToolsExist Condition="Exists($(ILDAsmFileLocation)) AND Exists($(ILAsmFileLocation))">true</ILToolsExist>
+  </PropertyGroup>
   
-  <UsingTask TaskName="FindILDAsm" TaskFactory="CodeTaskFactory" AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll">
-    <ParameterGroup>
-      <ILDAsmFileLocation ParameterType="System.String" Output="true" />
-    </ParameterGroup>
-    <Task>
-      <Using Namespace="Microsoft.Build.Utilities" />
-      <Code Type="Fragment" Language="cs">
-        <![CDATA[
-          ILDAsmFileLocation = Microsoft.Build.Utilities.ToolLocationHelper.GetPathToDotNetFrameworkSdkFile("ildasm.exe" , TargetDotNetFrameworkVersion.Latest);
-        ]]>
-      </Code>
-    </Task>
-  </UsingTask>
-  <Target Name="GeneratePortablePIA" BeforeTargets="SignFiles" AfterTargets="Build">
-    <FindILDAsm>
-      <Output PropertyName="ILDAsmFileLocation" TaskParameter="ILDAsmFileLocation" />
-    </FindILDAsm>
+  <Target Name="GeneratePortablePIA" BeforeTargets="SignFiles" AfterTargets="Build" Inputs="@(MakePIAPortableFiles)" Outputs="@(MakePIAPortableFiles->'$(PIAOutput)\%(Filename)%(Extension)')">
     <Message Importance="High" Text="ILDAsmFileLocation: $(ILDAsmFileLocation)" />
-    <Error Condition="'$(ILDAsmFileLocation)' == ''" Text="Unable to locate ILDasm.exe" />
-    <Exec Command="$(OutputPath)MakePortableLegacyPIAS.cmd &quot;$(ILDAsmFileLocation)&quot; &quot;$(PIAOutput)&quot;" />
+    <Message Importance="High" Text="ILAsmFileLocation: $(ILAsmFileLocation)" />
+    <Error Condition="'$(ILToolsExist)' != 'true'" Text="Unable to retrieve IL tools needed for MakePIAPortable" />
+
+    <!-- On linux, make MakePIAPortable exe runnable -->
+    <Exec Command="chmod 700 $(MakePIAPortableExe)" WorkingDirectory="$(OutputPath)" Condition="$([MSBuild]::IsOSPlatform('linux'))" />
+
+    <!-- Clean and create drop directory -->
+    <RemoveDir Directories="$(PIAOutput)" />
+    <MakeDir Directories="$(PIAOutput)" />
+
+    <!-- Run ildasm -> MakePIAPortable -> ilasm -->
+    <Exec Command="$(ILDAsmFileLocation) ./%(MakePIAPortableFiles.Identity) -OUT=./drop/%(MakePIAPortableFiles.Filename).il" WorkingDirectory="$(OutputPath)" />
```

Comment:
>ILDAsmFileLocation [](http://example.com/codeflow?start=21&length=18)

Does the nuget package version also support `/NOBAR` (disable the progress bar).

Reply:
I believe they got rid of the progress bar.

```
%USER%\.nuget\packages\runtime.win-x64.microsoft.netcore.ildasm\5.0.0\runtimes\win-x64\native\ildasm.exe /NOBAR
Microsoft (R) .NET IL Disassembler.  Version 5.0.0
Copyright (c) Microsoft Corporation.  All rights reserved.

INVALID COMMAND LINE OPTION: /NOBAR

Usage: ildasm [options] <file_name> [options]

Options for output redirection:
  /OUT=<file name>    Direct output to file rather than to console.
  /HTML               Output in HTML format (valid with /OUT option only).
  /RTF                Output in rich text format (valid with /OUT option only).
Options for file/console output:
  /BYTES              Show actual bytes (in hex) as instruction comments.
  /RAWEH              Show exception handling clauses in raw form.
  /TOKENS             Show metadata tokens of classes and members.
  /SOURCE             Show original source lines as comments.
  /LINENUM            Include references to original source lines.
  /VISIBILITY=<vis>[+<vis>...]    Only disassemble the items with specified
          visibility. (<vis> = PUB | PRI | FAM | ASM | FAA | FOA | PSC)
  /PUBONLY            Only disassemble the public items (same as /VIS=PUB).
  /QUOTEALLNAMES      Include all names into single quotes.
  /NOCA               Suppress output of custom attributes.
  /CAVERBAL           Output CA blobs in verbal form (default - in binary form).
The following options are valid for file/console output only:
Options for EXE and DLL files:
  /UTF8               Use UTF-8 encoding for output (default - ANSI).
  /UNICODE            Use UNICODE encoding for output.
  /NOIL               Suppress IL assembler code output.
  /FORWARD            Use forward class declaration.
  /TYPELIST           Output full list of types (to preserve type ordering in round-trip).
  /PROJECT            Display .NET projection view if input is a .winmd file.
  /HEADERS            Include file headers information in the output.
  /ITEM=<class>[::<method>[(<sig>)]  Disassemble the specified item only

  /STATS              Include statistics on the image.
  /CLASSLIST          Include list of classes defined in the module.
  /ALL                Combination of /HEADER,/BYTES,/STATS,/CLASSLIST,/TOKENS

Options for EXE,DLL,OBJ and LIB files:
  /METADATA[=<specifier>] Show MetaData, where <specifier> is:
          MDHEADER    Show MetaData header information and sizes.
          HEX         Show more things in hex as well as words.
          CSV         Show the record counts and heap sizes.
          UNREX       Show unresolved externals.
          SCHEMA      Show the MetaData header and schema information.
          RAW         Show the raw MetaData tables.
          HEAPS       Show the raw heaps.
          VALIDATE    Validate the consistency of the metadata.
Options for LIB files only:
  /OBJECTFILE=<obj_file_name> Show MetaData of a single object file in library

Option key is '-' or '/', options are recognized by first 3 characters

Example:  ildasm /tok /byt myfile.exe /out=myfile.il
```